skyportal/handlers/api/phot_stat.py

```diff
@@ -187,12 +197,521 @@ def delete(self, obj_id=None):
+            schema:
+              type: integer
+            description: Page number for iterating through all sources. Defaults to 1
+          - in: query
+            name: createdAtStartTime
+            required: false
+            schema:
+              type: string
+            description: |
+              arrow parseable string, only objects
+              that have been created after this time
+              will be checked for missing PhotStats.
+          - in: query
+            name: createdAtEndTime
+            required: false
+            schema:
+              type: string
+            description: |
+              arrow parseable string, only objects
+              that have been created before this time
+              will be checked for missing PhotStats.
+        responses:
+            200:
+              content:
+                application/json:
+                  schema:
+                    allOf:
+                      - $ref: '#/components/schemas/Success'
+                      - type: object
+                        properties:
+                          data:
+                            type: object
+                            properties:
+                              totalMatches:
+                                type: integer
+                              pageNumber:
+                                type: integer
+                              numPerPage:
+                                type: integer
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        try:
+            page_number = int(self.get_query_argument('pageNumber', 1))
+            num_per_page = min(
+                int(self.get_query_argument("numPerPage", DEFAULT_SOURCES_PER_PAGE)),
+                MAX_SOURCES_PER_PAGE,
+            )
+        except ValueError:
+            return self.error(
+                f'Cannot parse inputs pageNumber ({page_number}) '
+                f'or numPerPage ({num_per_page}) as an integers.'
+            )
+
+        created_at_start_time = self.get_query_argument('createdAtStartTime', None)
+        created_at_end_time = self.get_query_argument('createdAtEndTime', None)
+
+        with DBSession() as session:
+            stmt = sa.select(Obj).options(joinedload(Obj.photstats))
+            try:
+                if created_at_start_time:
+                    created_at_start_time = str(
+                        arrow.get(created_at_start_time.strip()).datetime
+                    )
+                    stmt = stmt.where(Obj.created_at >= created_at_start_time)
+                if created_at_end_time:
+                    created_at_end_time = str(
+                        arrow.get(created_at_end_time.strip()).datetime
+                    )
+                    stmt = stmt.where(Obj.created_at <= created_at_end_time)
+            except arrow.parser.ParserError:
+                return self.error(
```

Comment:
I assume leaving this DBSession context will properly shutdown the session?

Reply:
That's what it is supposed to do, an anyway I don't think we are changing anything in the DB until after these return statements.